napari/layers/labels/labels.py

```diff
@@ -460,27 +460,60 @@ def color(self):
 
     @color.setter
     def color(self, color):
-
         if not color:
             color = {}
-            color_mode = LabelColorMode.AUTO
-        else:
-            color_mode = LabelColorMode.DIRECT
 
         if self._background_label not in color:
             color[self._background_label] = 'transparent'
-
         if None not in color:
             color[None] = 'black'
 
         colors = {
             label: transform_color(color_str)[0]
             for label, color_str in color.items()
         }
-
         self._color = colors
+
+        # if colors just contains default colors for None and background
```

Comment:
... then?

It would be good to elaborate in this comment so that future generations can understand what this code is doing. I had to go and read the two issues to figure out why this is needed. Both here and in the test above, please reference the issues.